src/com/goide/runconfig/application/GoApplicationRunConfiguration.java

```diff
@@ -17,32 +17,33 @@
 package com.goide.runconfig.application;
 
 import com.goide.runconfig.GoModuleBasedConfiguration;
-import com.goide.runconfig.GoRunConfigurationWithMain;
-import com.goide.runconfig.ui.GoRunConfigurationEditorForm;
+import com.goide.runconfig.GoRunConfigurationBase;
+import com.goide.runconfig.ui.GoApplicationRunConfigurationEditorForm;
 import com.intellij.execution.configurations.ConfigurationType;
 import com.intellij.execution.configurations.ModuleBasedConfiguration;
 import com.intellij.execution.configurations.RunConfiguration;
+import com.intellij.execution.configurations.RuntimeConfigurationException;
 import com.intellij.execution.runners.ExecutionEnvironment;
 import com.intellij.openapi.module.Module;
 import com.intellij.openapi.options.SettingsEditor;
 import com.intellij.openapi.project.Project;
 import org.jetbrains.annotations.NotNull;
 
-public class GoApplicationConfiguration extends GoRunConfigurationWithMain<GoApplicationRunningState> {
-  public GoApplicationConfiguration(Project project, String name, @NotNull ConfigurationType configurationType) {
+public class GoApplicationRunConfiguration extends GoRunConfigurationBase<GoApplicationRunningState> {
```

Comment:
Good point, I will clean all of this up tomorrow.  Thanks for the feedback, and let me know if you have something better in the works. This was all just a quick hack on my part and I can see that there is probably a much more elegant way to do this eventually.